packages/electron/src/runParity.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: MIT
+
+import { ChildProcess, spawn } from 'child_process';
+import { chmod } from 'fs';
+import { promisify } from 'util';
+
+import { getParityPath } from './getParityPath';
+import logCommand from './utils/logCommand';
+import logger from './utils/logger';
+
+interface RunParityOptions {
+  flags: string[];
+  onParityError: (error: Error) => void;
+}
+
+/**
+ * @ignore
+ */
+const fsChmod = promisify(chmod);
+
+/**
+ * @ignore
+ */
+let parity: ChildProcess = null; // Will hold the running parity instance
+
+/**
+ * These are errors output by parity, which we should ignore (i.e. don't
+ * panic). They happen when an instance of parity is already running, and
+ * parity-electron tries to launch another one.
+ *
+ * @ignore
+ */
+const catchableErrors = [
+  'is already in use, make sure that another instance of an Ethereum client is not running',
+  'IO error: While lock file:'
+];
+
+/**
+ * Spawns a child process to run Parity.
+ */
+export async function runParity(
+  options: RunParityOptions = {
```

Comment:
needs to be verified but I think that for example `onParityError` would be undefined if we just pass as argument to runParity `{flags: []}`

cf https://github.com/paritytech/fether/pull/163/commits/752ec5f4fc7172c84d2dc4fe1defe1d5f07d2f9f

Reply:
Ah yes, you're right.